### PR TITLE
Adjust _get_rooms_changed comments

### PR DESCRIPTION
**Pull request by DMRobertson**
C.f. https://github.com/matrix-org/synapse/pull/11494#pullrequestreview-827780886

### CODE REVIEW

**Comment by DMRobertson** [comment]
@richvdh hopefully this is a better job of my earlier attempt!

File: synapse/handlers/sync.py [hunk 1]
```diff
@@ -1662,20 +1662,20 @@ async def _get_rooms_changed(
...
+           `since_token` up to `now_token`).
+        2. Uses those to place the room in one of the four categories above.
+        3. Builds a `_RoomChanges` struct to record this, and return that struct.
 
-        - we include a "state_delta", to describe the changes in state over the gap,
-        - we include all membership events applying to the user making the request,
-          even those in the gap.
-
-        See the spec for the rationale:
-            https://spec.matrix.org/v1.1/client-server-api/#syncing
```
Comment (richvdh) [inline synapse/handlers/sync.py:1678]: It feels like the removed stuff was correct and maybe useful? YMMV though.

**Review by richvdh — APPROVED** [review]

File: synapse/handlers/sync.py (hunk 1 above)
Comment (DMRobertson) [reply synapse/handlers/sync.py:1678]: Hmm, maybe I can find a better place for it.

**Review by richvdh — APPROVED** [review]
lgtm!

**Comment by DMRobertson** [comment]
I'm going to assume that's a flakey test (https://github.com/matrix-org/sytest/issues/1180) and re-run with automerge.

**Pull request merged**